source/base/index_set.cc

```diff
@@ -232,7 +232,27 @@ IndexSet::get_view(const size_type begin, const size_type end) const
   return result;
 }
 
-
+std::vector<IndexSet>
+IndexSet::split_by_block(
+  const std::vector<types::global_dof_index> &dofs_per_block,
+  const IndexSet &                            whole_set)
+{
+  std::vector<IndexSet> partitioned;
+  const unsigned int    n_blocks = dofs_per_block.size();
+  partitioned.clear();
+
+  partitioned.resize(n_blocks);
```

Comment:
```suggestion
  partitioned.reserve(n_blocks);
```

Reply:
Sorry I should have used the direct commit